graph-builder/src/registry.rs

```diff
@@ -125,9 +125,20 @@ pub fn fetch_releases(
                 tag.to_owned(),
             )
         })
+        .filter_map(|release| release)
         .collect();
 
-    thread_runtime.block_on(releases)
+    thread_runtime.block_on(releases).and_then(|releases| {
+        if releases.is_empty() {
+            Err(format_err!(
```

Comment:
Since we're having a return type of `Result<Vec<Release>, Error>` and the function is named `fetch_releases` it seems we should explain why we were not able to deliver a `Vec<Release>`. An alternative would be to rename to `try_fetch_releases(..) -> Result<Option<Vec<Release>>, Error>` and return `Ok(None)` in this case.

Reply:
But we _were_ able to deliver the vector of releases. It just happened to be empty. If any single tag is allowed to omit the release metadata (and be excluded from the graph), why isn't ever tag allowed to omit the release metadata?

Reply:
> If any single tag is allowed to omit the release metadata (and be excluded from the graph), why isn't ever tag allowed to omit the release metadata?

Because the expectation towards the repository is that it can yield at least one release, not that every tag can yield a release; at least that's how I understood it from the OOB discussion with @smarterclayton. This being said, I don't have a very strong opinion on it which it is, as long as it's intuitive to read and use the code, i.e. the naming and return types set realistic expectations.

What do you think of my rename/type change suggestion from above?

Reply:
I went ahead and implemented your suggestion. What remains is us agreeing on the naming and return types ;-)